### ISSUE TITLE
Allow reset of constructor params

### ISSUE THREAD

**Issue by Devilsmaul** [gksander/CIE-ColorConverter#6]
Hey there,

actually i wanted to implement a library based on Bruce Lindblooms calculations pages and then i found your **fantastic** implementation!

I would like to use it in my NodeRed node so i would have to change more often the parameters like RefWhite or RGBModel.
What do you think about adding setters to the parameters? This would make it even faster because i do not have to instantiate a new converter.
And could you please add a LICENCE file to your repository?  =)

**Comment by gksander** [comment]
Hi @Devilsmaul – I just added a LICENSE file to the repo, thanks for catching that!

As for the parameters, I suppose I could add formal setters for those. In the meantime though, those parameters are actually public properties on the converter instance. You can actually set those right now (they have the same name as the constructor parameter names). See below.

```ts
import { ColorConverter } from "cie-colorconverter";

const c = new ColorConverter();

// ... do stuff ...

c.RefWhite = "D55";

// ... do more stuff ...
```

**Comment by gksander** [comment]
An update on this: I just did a big rewrite of the library to move to a function-based approach. Each conversion method is now a standalone function that can called individually – which should be more performant, and things are less tangled up.

**Comment by Devilsmaul** [comment]
Awesome! Thanks!

One more: Could you also export your Options, and Enum? Otherwise i would have to redeclare them again if i want to use these types
`
export type NumericTriple = [number, number, number];
declare type RefWhiteType = "A" | "B" | "C" | "D50" | "D55" | "D65" | "D75" | "E" | "F2" | "F7" | "F11";
declare type RGBModelType =
  | "Adobe RGB (1998)"
  | "AppleRGB"
  | "Best RGB"
  | "Beta RGB"
  | "Bruce RGB"
  | "CIE RGB"
  | "ColorMatch RGB"
  | "Don RGB 4"
  | "ECI RGB v2"
  | "Ekta Space PS5"
  | "NTSC RGB"
  | "PAL/SECAM RGB"
  | "ProPhoto RGB"
  | "SMPTE-C RGB"
  | "sRGB"
  | "Wide Gamut RGB";
declare type GammaModelType = "1.0" | "1.8" | "2.2" | "sRGB" | "L*";
declare type AdaptationType = "Bradford" | "von Kries" | "XYZ Scaling" | "None";
declare type Options = {
  refWhite?: RefWhiteType;
  rgbModel?: RGBModelType;
  gammaModel?: GammaModelType;
  adaptation?: AdaptationType;
};`

**Comment by gksander** [comment]
@Devilsmaul yep, easy enough! Just exported some of the option types (available in version `3.1.0`).

Also worth pointing out that you could extract these types using some TS utility types like `Parameters`, `NonNullable`, etc.. For example – you could get the converter options by doing something like:

```ts
import { RGBToXYZ, ConverterOptions } from "cie-colorconverter";

type DerivedNumericTriple = NonNullable<Parameters<typeof RGBToXYZ>[0]>;
type DerivedOptions = NonNullable<Parameters<typeof RGBToXYZ>[1]>;
// ConverterOptions and DerivedOptions are the same type.
```

This is a super-convenient trick for grabbing types from libraries that don't directly export all of their types.
